beater/pub.go

```diff
@@ -66,30 +68,30 @@ func (p *publisher) Stop() {
 	p.m.Lock()
 	p.stopped = true
 	p.m.Unlock()
-	close(p.events)
+	close(p.payloads)
 	p.client.Close()
 }
 
 // Send tries to forward events to the publishers worker. If the queue is full,
 // an error is returned.
 // Calling send after Stop will return an error.
-func (p *publisher) Send(batch []beat.Event) error {
+func (p *publisher) Send(payload pr.Payload) error {
 	p.m.RLock()
 	defer p.m.RUnlock()
 	if p.stopped {
-		return errChanneClosed
+		return errChannelClosed
 	}
 
 	select {
-	case p.events <- batch:
+	case p.payloads <- payload:
 		return nil
 	case <-time.After(time.Second * 1): // this forces the go scheduler to try something else for a while
 		return errFull
 	}
 }
 
 func (p *publisher) run() {
-	for batch := range p.events {
-		p.client.PublishAll(batch)
+	for payload := range p.payloads {
+		p.client.PublishAll(payload.Transform())
```

Comment:
This effectively makes transformation single-threaded. In a multi-core machine, we should be able to utilize multiple cores for transformation. 

A way to solve this could be to start multiple go routines (for example, as many as there are cores. We can get the core-count by using `runtime.NumCPU()`). 

Something to be aware of: As far as I remember, beats `Client`s are not safe to use from multiple go routines. Perhaps each go routine could just have its own `Client`.

Reply:
Looking into the client code at beats, `PublishAll` uses a `c.mutex.Lock()`, so calling this from multiple go routines is not an issue:
https://github.com/elastic/beats/blob/master/libbeat/publisher/pipeline/client.go#L34